plugins/module_utils/_api_data.py

```diff
@@ -9,8 +9,67 @@
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
+from ansible_collections.community.routeros.plugins.module_utils.version import LooseVersion
+
+
+def _compare(a, b, comparator):
+    if comparator == '==':
+        return a == b
+    if comparator == '!=':
+        return a != b
+    if comparator == '<':
+        return a < b
+    if comparator == '<=':
+        return a <= b
+    if comparator == '>':
+        return a > b
+    if comparator == '>=':
+        return a >= b
+    raise ValueError('Unknown comparator "{comparator}"'.format(comparator=comparator))
+
 
 class APIData(object):
+    def __init__(self,
+                 unversioned=None,
+                 versioned=None):
+        if (unversioned is None) == (versioned is None):
+            raise ValueError('either unversioned or versioned must be provided')
+        self.unversioned = unversioned
+        self.versioned = versioned
+        if self.unversioned is not None:
+            self.needs_version = False
+            self.fully_understood = self.unversioned.fully_understood
+        else:
+            self.needs_version = self.versioned is not None
+            # Mark as 'fully understood' if it is for at least one version
+            self.fully_understood = False
+            for dummy, dummy, unversioned in self.versioned:
+                if unversioned.fully_understood:
+                    self.fully_understood = True
+                    break
+
+    def provide_version(self, version):
+        if not self.needs_version:
+            return self.unversioned.fully_understood
+        api_version = LooseVersion(version)
+        for other_version, comparator, unversioned in self.versioned:
+            if other_version == '*' and comparator == '*':
+                self.unversioned = unversioned
+                return self.unversioned.fully_supported
```

Comment:
I think the returned property should be `fully_understood`. There's no `fully_support`.

> AttributeError: 'VersionedAPIData' object has no attribute 'fully_supported'